skimage/segmentation/_join.py

```diff
@@ -114,20 +116,29 @@ def relabel_sequential(label_field, offset=1):
     >>> relab
     array([5, 5, 6, 6, 7, 9, 8])
     """
+    offset = int(offset)
+    if offset <= 0:
+        raise ValueError("Offset must be strictly positive.")
+    if np.min(label_field) < 0:
+        raise ValueError("Cannot relabel array that contains negative values.")
     m = label_field.max()
-    if not np.issubdtype(label_field.dtype, np.signedinteger):
+    if not np.issubdtype(label_field.dtype, np.integer):
         new_type = np.min_scalar_type(int(m))
         label_field = label_field.astype(new_type)
         m = m.astype(new_type)  # Ensures m is an integer
     labels = np.unique(label_field)
     labels0 = labels[labels != 0]
-    if m == len(labels0):  # nothing to do, already 1...n labels
```

Comment:
Btw, the previous line
```python
if m == len(labels0):  # nothing to do, already 1...n labels
```
was only valid for `offset = 1`. This was only mentioned in the comment but not checked, i.e. it should've been
```python
if offset == 1 and m == len(labels0):  # nothing to do, already 1...n labels
```
Anyway, I guess hardly anyone uses `offset != 1`. Why was the previous function `relabel_from_one` replaced with `relabel_sequential`? (Since the `offset` introduces quite a bit of (subtle) complexity.)

Also, `relabel_sequential` assumes that the background has value 0, but I noticed `color.label2rgb` assumes (by default) that the background label has value -1.

Reply:
> Why was the previous function `relabel_from_one` replaced with `relabel_sequential`?

Someone requested the functionality and it seemed like the right thing to do and an easy fix. Oops. =)

> Also, `relabel_sequential` assumes that the background has value 0, but I noticed `color.label2rgb` assumes (by default) that the background label has value -1.

Yes, historically, skimage used -1 as the background label, but we have slowly started homogenising to 0. But this will take time.

Reply:
Good to know, thanks!